jina/types/arrays/document.py

```diff
@@ -375,14 +375,19 @@ def reverse(self):
             hi_idx -= 1
         self._update_id_to_index_map()
 
-    def sort(self, key=None, *args, **kwargs):
+    def sort(self, key=None, top_k: int = None, *args, **kwargs):
         """
         Sort the items of the :class:`DocumentArray` in place.
 
         :param key: key callable to sort based upon
+        :param top_k: sort only upto top_k Document from DocumentArray
         :param args: variable set of arguments to pass to the sorting underlying function
         :param kwargs: keyword arguments to pass to the sorting underlying function
         """
+        # Only for DocumentArray
+        if isinstance(self._pb_body, list):
```

Comment:
Maybe it was not clear from the documentation. The point is to only sort the `first top_k` but of the sorted list.

There are sorting algorithms that do not sort a complete collection but that only guarantee that they first `top_k` elements are sorted while the rest may not be